features/flows/classification/engine/impl/src/main/java/org/opennms/netmgt/flows/classification/internal/value/IpValue.java

```diff
@@ -67,26 +71,149 @@ private void parse(final StringValue input) {
+                ranges.add(new IpAddressRange(rangedValues.get(0), rangedValues.get(1)));
             } else {
-                verifyIpAddress(eachValue);
-                ranges.add(new IPAddressRange(eachValue.getValue(), eachValue.getValue()));
+                ranges.add(new IpAddressRange(eachValue));
             }
         }
     }
 
     public boolean isInRange(final String address) {
-        return ranges.stream().anyMatch(r -> r.contains(address));
+        return ranges.stream().anyMatch(r -> r.isInRange(address));
     }
 
-    private static void verifyIpAddress(final StringValue stringValue) {
-        Objects.requireNonNull(stringValue);
-        if (!InetAddresses.isInetAddress(stringValue.getValue())) {
-            throw new IllegalArgumentException("Provided ip address '" + stringValue.getValue() + "' is invalid");
+    private static class IpAddressRange {
+
+        private final IpAddressMatcher matcher;
+
+        private IpAddressRange(final StringValue from, final StringValue to) {
+            this.matcher = new IpAddressRangeMatcher(from.getValue(), to.getValue());
+        }
+
+        private IpAddressRange(final StringValue eachValue) {
+            if (eachValue.contains("/")) {
+                this.matcher = new IpV6CidrExpressionMatcher(eachValue.getValue());
+            } else {
+                this.matcher = new IpAddressRangeMatcher(eachValue.getValue());
+            }
+        }
+
+        public boolean isInRange(String input) {
+            return matcher.matches(input);
+        }
+    }
+
+    private interface IpAddressMatcher {
+        boolean matches(String address);
+    }
+
+    private static class IpAddressRangeMatcher implements IpAddressMatcher {
+
+        private final IPAddressRange range;
+
+        private IpAddressRangeMatcher(final String from, final String to) {
+            verifyIpAddress(from);
+            verifyIpAddress(to);
+            this.range = new IPAddressRange(from, to);
+        }
+
+        public IpAddressRangeMatcher(String value) {
+            verifyIpAddress(value);
+            this.range = new IPAddressRange(value);
+        }
+
+        @Override
+        public boolean matches(String address) {
+            return range.contains(address);
+        }
+
+        private static void verifyIpAddress(final String value) {
+            Objects.requireNonNull(value);
+            if (!InetAddresses.isInetAddress(value)) {
+                throw new IllegalArgumentException("Provided ip address '" + value + "' is invalid");
+            }
+        }
+    }
+
+    // Inspired by spring-security-web's IpAddressMatcher
+    public final static class IpV6CidrExpressionMatcher implements IpAddressMatcher {
+
+        private final int nMaskBits;
+        private final InetAddress requiredAddress;
+
+        /**
+         * Takes a specific IP address or a range specified using the
+         * IP/Netmask (e.g. 192.168.1.0/24 or 202.24.0.0/14).
```

Comment:
The examples should be v6 address or the class name is misleading

Reply:
Yes, that is true.